### ISSUE TITLE
[Bug 1.45] Unable to view commits from an oldish branch (even when filtered)

### ISSUE THREAD

**Issue by geordiemhall** [ForkIssues/TrackerWin#663]
Just updated to v1.45 and was excited by branch filtering performance improvements (thanks!), though seems like there may have been some regressions.

When working on large repositories like UE4, there can be a lot of upstream commits coming in daily. Fork seems to only show the last X commits in the graph, presumably for performance reasons. I'm not sure exactly what this limit is (1000?), but I've noticed it in previous versions of Fork. This meant that if you wanted to view a branch whose latest commit so old it wasn't shown in the graph view, you had to turn on the branch filter so that its latest commit would be at the top of the view (though ideally that workaround wouldn't be necessary at all).

But now in v1.45, if you have a branch filtered whose latest commit is old enough that it wouldn't fit within those top X commits shown in the graph view, you just get an empty white screen (i.e. no commits shown in the graph) and the commit "details" you're viewing don't update.

![image](https://user-images.githubusercontent.com/3511383/74599095-c498f680-50d0-11ea-9660-f930f2ae2024.png)

I've noticed that navigating to an old commit via Blame or File History dialogs also shows the same empty graph view, so probably affects the viewing of all old-ish commits.

Happy to provide more details if needed, and I can recommend using the UE4 repository as a good stress test for the graph, since you only have to scroll down a little before the diagram lines get super wide.

![image](https://user-images.githubusercontent.com/3511383/74599060-46d4eb00-50d0-11ea-8dfc-c08b44218ffe.png)

**Comment by DanPristupov** [comment]
Thank you for reporting this problem. This is the case I absolutely missed. Currently Fork loads `25000` commits (which I expected to be enough for everyone). If your branch is out of this range, it will not be visible. I can't fix this quickly, but I will be thinking how to improve the behavior (most probably we need to start loading commits dynamically).

As for now, I can introduce an option in increase this limit in the next update. Or can you rebase the old branch on a newer commit to make it visible?

**Comment by geordiemhall** [comment]
Gee UE4 must be even more active than I thought if the limit is that high, I was only trying to view commits from November last year!

For now I've just gone back to using Fork v1.44 since I couldn't be bothered rebasing just to view some commits on an experimental branch which will probably have a tonne of conflicts by now, but might try rebasing eventually :)

A configurable limit could help, although if it lets users slow their app to a crawl then might be a double-edged sword? Loading commits dynamically sounds like it'd be the nicest solution, even if it's just "try loading 10,000 at a time until the commit they requested is visible". In the meantime even just showing a warning message when the commit you're trying to view is past the 25,000 could be good (instead of a white screen/doing nothing).

The Unreal Engine repo currently has 123,000 commits, so could be useful for your testing! You can sign up for repo access for free from here if you want https://www.unrealengine.com/en-US/ue4-on-github . They also have some possibly long-running post checkout hooks to download binary dependencies etc. in case you ever wanted to test that side of things too

**Comment by geordiemhall** [comment]
Sorry to open this again, but after switching branches I seem to have gotten myself into a state where commits on master aren't even visible anymore.

With the UnrealEngine repo again - I was on `master`, then switched to `dev-rendering`, which is a different development "stream", which may well have diverged from master a super long time ago (so may well have had 25,000 unique commits on it?).

I was able to view commits on `dev-rendering` fine, but then switching back to `master` it seems that it may have blown past some limit and is now refusing to load commits for the tree view, even with filters applied etc.

Filtering the view to `dev-rendering` will show commits again, so I'm guessing by checking out the local branch it somehow overloaded Fork's commit cache?

Is there anything I can do to purge those other commits so I can view `master` commits again? I tried looking in `.git/fork-settings` in case there was anything I could reset but didn't see like much that'd affect this. 

![image](https://user-images.githubusercontent.com/3511383/89846859-b6cab680-dbc5-11ea-8d7b-78fff1a040bf.png)

Cheers!
